VideoLocker/src/main/java/org/edx/mobile/view/adapters/DiscussionCommentsAdapter.java

```diff
@@ -114,57 +123,53 @@ public void onBindViewHolder(RecyclerView.ViewHolder viewHolder, int position) {
-            } else {
-                backgroundRes = R.drawable.row_discussion_comment_background;
-            }
 
             iconDrawable = new IconDrawable(context, FontAwesomeIcons.fa_flag)
                     .sizeRes(context, R.dimen.edx_xxx_small)
-                    .colorRes(context, discussionComment.isAbuseFlagged() ? R.color.edx_brand_primary_base : R.color.edx_grayscale_neutral_dark);
+                    .colorRes(context, discussionComment.isAbuseFlagged() ? R.color.edx_brand_primary_base : R.color.edx_grayscale_neutral_base);
             holder.discussionCommentCountReportTextView.setText(discussionComment.isAbuseFlagged() ? context.getString(R.string.discussion_responses_reported_label) : context.getString(R.string.discussion_responses_report_label));
-            holder.discussionCommentCountReportTextView.setTextColor(context.getResources().getColor(R.color.edx_grayscale_neutral_dark));
+            holder.discussionCommentCountReportTextView.setTextColor(context.getResources().getColor(R.color.edx_grayscale_neutral_base));
 
             holder.discussionCommentCountReportTextView.setOnClickListener(new View.OnClickListener() {
                 public void onClick(final View v) {
                     listener.reportComment(discussionComment);
                 }
             });
         }
-        TextViewCompat.setCompoundDrawablesRelativeWithIntrinsicBounds(holder.discussionCommentCountReportTextView, iconDrawable, null, null, null);
-        holder.discussionCommentRow.setLayoutParams(layoutParams);
-        holder.discussionCommentRow.setBackgroundResource(backgroundRes);
-
-        String commentBody = discussionComment.getRawBody();
-        holder.discussionCommentBody.setText(commentBody);
 
-        DiscussionTextUtils.setAuthorAttributionText(holder.discussionCommentAuthorTextView,
-                DiscussionTextUtils.AuthorAttributionLabel.POST,
+        holder.authorLayoutViewHolder.populateViewHolder(config, discussionComment,
                 discussionComment, initialTimeStampMs,
                 new Runnable() {
                     @Override
                     public void run() {
                         listener.onClickAuthor(discussionComment.getAuthor());
                     }
                 });
+
+        TextViewCompat.setCompoundDrawablesRelativeWithIntrinsicBounds(
+                holder.discussionCommentCountReportTextView, iconDrawable, null, null, null);
+
+        String commentBody = discussionComment.getRawBody();
+        holder.discussionCommentBody.setText(commentBody);
     }
 
     @Override
     public int getItemCount() {
         int total = 1 + discussionComments.size();
-        if (progressVisible)
+        if (progressVisible) {
             total++;
+        }
         return total;
     }
 
     @Override
     public int getItemViewType(int position) {
+        if (position == 0) {
+            return RowType.RESPONSE;
+        }
```

Comment:
It would be better to move this case after the progress bar row check, since we might conceivably load the response from the server as well in the future.